src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private MyLinkedList.Node<T> first;
+    private MyLinkedList.Node<T> last;
+
+    MyLinkedList.Node<T> node(int index) {
+        Node<T> foundedNode;
+        if (index < (size / 2)) {
+            foundedNode = first;
+            for (int i = 0; i < index; i++) {
+                foundedNode = foundedNode.next;
+            }
+        } else {
+            foundedNode = last;
+            for (int i = size - 1; i > index; i--) {
+                foundedNode = foundedNode.prev;
+            }
+        }
+        return foundedNode;
+    }
+
+    private void checkElementIndex(int index) {
+        if (!(index >= 0 && index < size)) {
+            throw new IndexOutOfBoundsException("Index: " + index + " is more then size " + size);
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("Index: " + index + " is more then size " + size);
+        }
+    }
+
+    void unlink(Node<T> removedNode) {
+        final MyLinkedList.Node<T> next = removedNode.next;
```

Comment:
```suggestion
        Node<T> next = removedNode.next;
```
Doesn't it work that way? Also don't use final in methods